### ISSUE TITLE
`amazon/dynamodb-local:2.5.0` fails to start

### ISSUE THREAD

**Issue by Maxwell2022** [awslabs/amazon-dynamodb-local-samples#18]
https://github.com/amazonlinux/container-images/issues/108

### Product

Not Specific / Don't Know

### What is your question?

I noticed this new version of `amazon/dynamodb-local` `v2.5.0` (18 hours ago).
This version fails to start on Apple Silicone (and other OS apparently) with the following error in the container:

```
2024-05-29 17:05:23 Initializing DynamoDB Local with the following configuration:
2024-05-29 17:05:23 Port:       8000
2024-05-29 17:05:23 InMemory:   false
2024-05-29 17:05:23 Version:    2.5.0
2024-05-29 17:05:23 DbPath:     ./data
2024-05-29 17:05:23 SharedDb:   true
2024-05-29 17:05:23 shouldDelayTransientStatuses:       false
2024-05-29 17:05:23 CorsParams: null
2024-05-29 17:05:23 
2024-05-29 17:05:39 May 29, 2024 7:05:38 AM com.almworks.sqlite4java.Internal log
2024-05-29 17:05:39 WARNING: [sqlite] SQLiteDBAccess$1@21e7144f: job exception
2024-05-29 17:05:39 com.almworks.sqlite4java.SQLiteException: [1] DB[1] prepare() ALTER TABLE dm ADD COLUMN MaxReadRequestUnits INTEGER DEFAULT 0ADD COLUMN MaxWriteRequestUnits INTEGER DEFAULT 0; [unrecognized token: "0ADD"]
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteConnection.throwResult(SQLiteConnection.java:1436)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteConnection.prepare(SQLiteConnection.java:580)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteConnection.prepare(SQLiteConnection.java:635)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteConnection.prepare(SQLiteConnection.java:622)
2024-05-29 17:05:39     at com.amazonaws.services.dynamodbv2.local.shared.access.sqlite.AmazonDynamoDBOfflineSQLiteJob.getPreparedStatement(AmazonDynamoDBOfflineSQLiteJob.java:159)
2024-05-29 17:05:39     at com.amazonaws.services.dynamodbv2.local.shared.access.sqlite.SQLiteDBAccess$1.doWork(SQLiteDBAccess.java:531)
2024-05-29 17:05:39     at com.amazonaws.services.dynamodbv2.local.shared.access.sqlite.SQLiteDBAccess$1.doWork(SQLiteDBAccess.java:406)
2024-05-29 17:05:39     at com.amazonaws.services.dynamodbv2.local.shared.access.sqlite.AmazonDynamoDBOfflineSQLiteJob.job(AmazonDynamoDBOfflineSQLiteJob.java:117)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteJob.execute(SQLiteJob.java:372)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteQueue.executeJob(SQLiteQueue.java:534)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteQueue.queueFunction(SQLiteQueue.java:667)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteQueue.runQueue(SQLiteQueue.java:623)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteQueue.access$000(SQLiteQueue.java:77)
2024-05-29 17:05:39     at com.almworks.sqlite4java.SQLiteQueue$1.run(SQLiteQueue.java:205)
2024-05-29 17:05:39     at java.base/java.lang.Thread.run(Thread.java:840)
2024-05-29 17:05:39 
```

version `2.4.0` has no problem, starts without errors:

```
2024-05-29 17:19:26 Initializing DynamoDB Local with the following configuration:
2024-05-29 17:19:26 Port:       8000
2024-05-29 17:19:26 InMemory:   false
2024-05-29 17:19:26 Version:    2.4.0
2024-05-29 17:19:26 DbPath:     ./data
2024-05-29 17:19:26 SharedDb:   true
2024-05-29 17:19:26 shouldDelayTransientStatuses:       false
2024-05-29 17:19:26 CorsParams: null
2024-05-29 17:19:26 
```

**Comment by pradeep-maira** [comment]
Faced the same issue on linux as well. Changing the version to 2.4.0 resolved the issue.

**Comment by shetsa-amzn** [comment]
Thank you for reporting this bug. We are currently trying to replicate this issue on our end and looking for the bug. We will get back to you with our investigation. In the meantime, I assume you are using the DynamoDB Local Docker Image setup and have used the `-sharedDb`option with a database that was already created on a `DbPath` from a previous version. @Maxwell2022 @pradeep-maira

**Comment by Maxwell2022** [comment]
yes, sorry this is the docker compose config:

```docker
  datastore-dynamodb:
    image: amazon/dynamodb-local:2.5.0
    ports: [8000:8000]
    working_dir: /home/dynamodblocal
    user: root
    command: '-jar DynamoDBLocal.jar -sharedDb -dbPath ./data'
    volumes:
      - dynamodb-volume:/home/dynamodblocal/data
```

**Comment by asitmoharna** [comment]
> Thank you for reporting this bug. We are currently trying to replicate this issue on our end and looking for the bug. We will get back to you with our investigation. In the meantime, I assume you are using the DynamoDB Local Docker Image setup and have used the `-sharedDb`option with a database that was already created on a `DbPath` from a previous version. @Maxwell2022 @pradeep-maira

Thanks @shetsa-amzn! 

Deleting the local volume solved the issue for us. So, in your case @Maxwell2022 you should delete your local docker volume (`docker/dynamodb`) folder.

**Comment by Maxwell2022** [comment]
Yes, this assumption seems correct. It's the pre-existent volume that trigger this error.
I'm guessing it would be fine for most of us to remove the old volume and create a new one.

The following command is working fine for me:

```
docker run -v ./tmp:/home/dynamodblocal/data -w /home/dynamodblocal amazon/dynamodb-local:2.5.0 -jar DynamoDBLocal.jar -sharedDb -dbPath ./data
```

**Comment by shetsa-amzn** [comment]
I have released the fix in the latest release version 2.5.1 of DynamoDB Local @Maxwell2022 @asitmoharna @pradeep-maira

Sorry for the inconvenience!
